src/v/kafka/protocol/wire.h

```diff
@@ -72,6 +73,7 @@ class decoder {
         auto [i, _] = _parser.read_unsigned_varint();
         return i;
     }
+    double read_double() { return std::bit_cast<double>(read_int64()); }
```

Comment:
Would something like this be sufficient to cover [the spec](https://kafka.apache.org/protocol#:~:text=Represents%20a%20double%2Dprecision%2064%2Dbit%20format%20IEEE%20754%20value.%20The%20values%20are%20encoded%20using%20eight%20bytes%20in%20network%20byte%20order%20(big%2Dendian).)?
```c++
    float64_t read_float64() {
        static_assert(std::numeric_limits<float64_t>::is_iec559, "oops");
        return std::bit_cast<float64_t>(read_int64());
    }
```

Reply:
Nice